src/frontend/src/optimizer/plan_node/generic/agg.rs

```diff
@@ -105,8 +105,11 @@ impl<PlanRef: GenericPlanRef> Agg<PlanRef> {
             && !self.agg_calls.is_empty()
             && self.agg_calls.iter().all(|call| {
                 let agg_kind_ok = !matches!(call.agg_kind, agg_kinds::simply_cannot_two_phase!());
-                let order_ok = matches!(call.agg_kind, agg_kinds::result_unaffected_by_order_by!())
-                    || call.order_by.is_empty();
+                let order_ok = matches!(
+                    call.agg_kind,
+                    agg_kinds::result_unaffected_by_order_by!()
+                        | AggKind::Builtin(PbAggKind::ApproxPercentile)
+                ) || call.order_by.is_empty();
```

Comment:
Is this because of the nature that `ApproxPercentile` uses buckets to calculate the percentile? So that the strict order is not important? If so, I guess we can add `ApproxPercentile` to `agg_kinds::result_unaffected_by_order_by`.

Reply:
ORDER BY affects it and can't be simply ignored.
If `ORDER BY DESC` is present, we rewrite it:
```
original:
approx_percentile(q, relative_error) within group (order by v1 desc)

rewrite:
approx_percentile(1-q, relative_error) within group (order by v1 asc)
```

See the test in `agg.yaml`.

Reply:
Oh, I see.